Instant parsing fails when optional milliseconds are absent

See [instant](https://www.hl7.org/fhir/datatypes.html#instant).

Here is an example error:

```
com.google.gson.JsonParseException: Failed to parse Instant from string: "2025-02-18T07:33:21+00:00"
        at au.csiro.fhir.model.FhirInstantDeserializer.deserialize(FhirInstantDeserializer.java:59)
        at au.csiro.fhir.model.FhirInstantDeserializer.deserialize(FhirInstantDeserializer.java:38)
        at com.google.gson.TreeTypeAdapter.read(TreeTypeAdapter.java:58)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:93)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:172)
        at com.google.gson.Gson.fromJson(Gson.java:803)
        at com.google.gson.Gson.fromJson(Gson.java:768)
        at com.google.gson.Gson.fromJson(Gson.java:717)
        at com.google.gson.Gson.fromJson(Gson.java:689)
        at au.csiro.fhir.model.FhirJsonSupport.fromJson(FhirJsonSupport.java:53)
        at au.csiro.fhir.export.ws.AsynResponseHandler.lambda$produceFinalResponse$6(AsynResponseHandler.java:111)
        at java.base/java.util.Optional.flatMap(Optional.java:289)
        at au.csiro.fhir.export.ws.AsynResponseHandler.produceFinalResponse(AsynResponseHandler.java:111)
        at au.csiro.fhir.export.ws.AsynResponseHandler.handleResponse(AsynResponseHandler.java:69)
        at au.csiro.fhir.export.ws.AsynResponseHandler.handleResponse(AsynResponseHandler.java:45)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:223)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:165)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:140)
        at au.csiro.fhir.export.ws.BulkExportAsyncService.checkStatus(BulkExportAsyncService.java:81)
        at au.csiro.fhir.export.ws.BulkExportTemplate$PoolingState.handle(BulkExportTemplate.java:157)
        at au.csiro.fhir.export.ws.BulkExportTemplate.run(BulkExportTemplate.java:296)
        at au.csiro.fhir.export.ws.BulkExportTemplate.export(BulkExportTemplate.java:280)
        at au.csiro.fhir.export.BulkExportClient.doExport(BulkExportClient.java:330)
        at au.csiro.fhir.export.BulkExportClient.export(BulkExportClient.java:303)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
        at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.time.format.DateTimeParseException: Text '2025-02-18T07:33:21+00:00' could not be parsed at index 19
        at java.base/java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:2052)
        at java.base/java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1954)
        at au.csiro.fhir.model.FhirFormats.parseFhirInstant(FhirFormats.java:56)
        at au.csiro.fhir.model.FhirInstantDeserializer.deserialize(FhirInstantDeserializer.java:57)
        ... 35 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -23,7 +23,7 @@
   <modelVersion>4.0.0</modelVersion>
 
   <groupId>au.csiro.fhir</groupId>
-  <version>1.0.0</version>
+  <version>1.0.1-SNAPSHOT</version>
   <artifactId>bulk-export</artifactId>
   <packaging>jar</packaging>
 

--- a/src/main/java/au/csiro/fhir/model/FhirFormats.java
+++ b/src/main/java/au/csiro/fhir/model/FhirFormats.java
@@ -32,7 +32,7 @@ public class FhirFormats {
   private static final ZoneId UTC_ZONE_ID = ZoneId.of("UTC");
 
   private static final DateTimeFormatter FHIR_INSTANT_FORMAT = DateTimeFormatter
-      .ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX").withZone(UTC_ZONE_ID);
+      .ofPattern("yyyy-MM-dd'T'HH:mm:ss[.SSS]XXX").withZone(UTC_ZONE_ID);
 
   /**
    * Formats an {@link Instant} as a FHIR instant string.

--- a/src/main/java/au/csiro/fhir/model/FhirInstantDeserializer.java
+++ b/src/main/java/au/csiro/fhir/model/FhirInstantDeserializer.java
@@ -44,15 +44,7 @@ class FhirInstantDeserializer implements JsonDeserializer<Instant> {
       throw new JsonParseException("Failed to parse Instant from non-primitive: " + json);
     }
     final JsonPrimitive primitive = json.getAsJsonPrimitive();
-    if (primitive.isNumber()) {
-      return Instant.ofEpochMilli(primitive.getAsLong());
-    } else if (primitive.isString()) {
-      try {
-        return Instant.ofEpochMilli(Long.parseLong(primitive.getAsString()));
-      } catch (final NumberFormatException __) {
-        // Not a number
-        // Continue to try to parse as a FHIR instan.
-      }
+    if (primitive.isString()) {
       try {
         return parseFhirInstant(primitive.getAsString());
       } catch (final DateTimeParseException dex) {

--- a/src/test/java/au/csiro/fhir/export/BulkExportClientWiremockTest.java
+++ b/src/test/java/au/csiro/fhir/export/BulkExportClientWiremockTest.java
@@ -81,7 +81,7 @@ class BulkExportClientWiremockTest {
       ))
       .toString();
   public static final String BULK_EXPORT_NO_FILES_RESPONSE = new JSONObject()
-      .put("transactionTime", 4934344343L)
+      .put("transactionTime", "1970-02-27T12:39:04.343Z")
       .put("request", "http://localhost:8080/$export")
       .put("requiresAccessToken", false)
       .put("output", new JSONArray())
@@ -90,7 +90,7 @@ class BulkExportClientWiremockTest {
   public static String bulkExportResponse_3_files(
       @Nonnull final WireMockRuntimeInfo wmRuntimeInfo) {
     return new JSONObject()
-        .put("transactionTime", "4934344343")
+        .put("transactionTime", "1970-02-27T12:39:04.343Z")
         .put("request", "http://localhost:8080/$export")
         .put("requiresAccessToken", false)
         .put("output", new JSONArray()
@@ -589,7 +589,7 @@ class BulkExportClientWiremockTest {
         .whenScenarioStateIs("done")
         .willReturn(aResponse().withStatus(200).withBody(
             new JSONObject()
-                .put("transactionTime", 4934344343L)
+                .put("transactionTime", "1970-02-27T12:39:04.343Z")
                 .put("request", "http://localhost:8080/$export")
                 .put("requiresAccessToken", false)
                 .put("output", new JSONArray()
@@ -681,7 +681,7 @@ class BulkExportClientWiremockTest {
         .whenScenarioStateIs("done")
         .willReturn(aResponse().withStatus(200).withBody(
             new JSONObject()
-                .put("transactionTime", 4934344343L)
+                .put("transactionTime", "1970-02-27T12:39:04.343Z")
                 .put("request", "http://localhost:8080/$export")
                 .put("requiresAccessToken", false)
                 .put("output", new JSONArray()

--- a/src/test/java/au/csiro/fhir/model/FhirInstantDeserializerTest.java
+++ b/src/test/java/au/csiro/fhir/model/FhirInstantDeserializerTest.java
@@ -31,45 +31,23 @@ class FhirInstantDeserializerTest {
 
   final FhirInstantDeserializer fhirInstantDeserializer = new FhirInstantDeserializer();
 
-  final static Instant TEST_INSTANT = Instant.parse("2023-01-02T00:01:02.123Z");
-  final static long TEST_INSTANT_EPOCH_MILLI = TEST_INSTANT.toEpochMilli();
-
   @Test
-  void deserializeFromMillsecondsValueAsNumber() {
-    assertEquals(TEST_INSTANT,
-        fhirInstantDeserializer.deserialize(new JsonPrimitive(TEST_INSTANT_EPOCH_MILLI),
-            Instant.class,
-            null));
-  }
-
-  @Test
-  void deserializeFromMillsecondsValueAsString() {
-    assertEquals(TEST_INSTANT,
-        fhirInstantDeserializer.deserialize(
-            new JsonPrimitive(String.valueOf(TEST_INSTANT_EPOCH_MILLI)), Instant.class,
-            null));
-  }
-
-
-  @Test
-  void deserializeFromFHIRStringValueWithZoneZ() {
+  void deserializeFromStringWithZoneZ() {
     assertEquals(Instant.parse("1971-10-12T01:02:03.123Z"),
         fhirInstantDeserializer.deserialize(new JsonPrimitive("1971-10-12T01:02:03.123Z"),
-            Instant.class,
-            null));
+            Instant.class, null));
   }
 
   @Test
-  void deserializeFromFHIRStringValueWithExplicitOffset() {
-    assertEquals(TEST_INSTANT,
+  void deserializeFromStringWithExplicitOffset() {
+    assertEquals(Instant.parse("2023-01-02T00:01:02.123Z"),
         fhirInstantDeserializer.deserialize(new JsonPrimitive("2023-01-02T01:31:02.123+01:30"),
-            Instant.class,
-            null));
+            Instant.class, null));
   }
 
 
   @Test
-  void deserializeFromFHIRStringValueWithNoMilliseconds() {
+  void deserializeFromStringWithNoMilliseconds() {
     assertEquals(Instant.parse("2023-01-02T01:31:02Z"),
         fhirInstantDeserializer.deserialize(new JsonPrimitive("2023-01-02T01:31:02Z"),
             Instant.class, null));
@@ -87,6 +65,14 @@ class FhirInstantDeserializerTest {
     final JsonParseException ex = assertThrows(JsonParseException.class,
         () -> fhirInstantDeserializer.deserialize(new JsonArray(0), Instant.class, null));
     assertEquals("Failed to parse Instant from non-primitive: []", ex.getMessage());
+  }
+
+  @Test
+  void deserializeFromStringWithInsufficientPrecision() {
+    final JsonParseException ex = assertThrows(JsonParseException.class,
+        () -> fhirInstantDeserializer.deserialize(new JsonPrimitive("2023-01-02T01:31Z"),
+            Instant.class, null));
+    assertEquals("Failed to parse Instant from string: \"2023-01-02T01:31Z\"", ex.getMessage());
   }
 
 }

--- a/src/test/java/au/csiro/fhir/model/FhirInstantDeserializerTest.java
+++ b/src/test/java/au/csiro/fhir/model/FhirInstantDeserializerTest.java
@@ -17,14 +17,14 @@
 
 package au.csiro.fhir.model;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
 import com.google.gson.JsonArray;
 import com.google.gson.JsonParseException;
 import com.google.gson.JsonPrimitive;
-import org.junit.jupiter.api.Test;
-
 import java.time.Instant;
-
-import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
 
 class FhirInstantDeserializerTest {
 
@@ -66,7 +66,15 @@ class FhirInstantDeserializerTest {
             Instant.class,
             null));
   }
-  
+
+
+  @Test
+  void deserializeFromFHIRStringValueWithNoMilliseconds() {
+    assertEquals(Instant.parse("2023-01-02T01:31:02Z"),
+        fhirInstantDeserializer.deserialize(new JsonPrimitive("2023-01-02T01:31:02Z"),
+            Instant.class, null));
+  }
+
   @Test
   void deserializeFromInvalidPrimitive() {
     final JsonParseException ex = assertThrows(JsonParseException.class,

